File: NextPassAPI/NextPassAPI/Controllers/AuthController.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using NextPassAPI.Data.Models;
using NextPassAPI.Data.Models.Requests;
using NextPassAPI.Data.Models.Responses;
using NextPassAPI.Services.Interfaces;

namespace NextPassAPI.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class AuthController : ControllerBase
    {
        private readonly IAuthService _authService;

        public AuthController(IAuthService authService)
        {
            _authService = authService;
        }


        [HttpPost("login")]
        public async Task<IActionResult> LoginUser([FromBody] AuthRequest userLoginRequest)
        {
            try
            {
                var user = await _authService.LoginUser(userLoginRequest.Email, userLoginRequest.Password);
                if (user == null)
                {
                    var emptyResponse = new ApiResponse<User>(false, "Invalid email or password", null);
                    return Unauthorized(emptyResponse);
                }
                var response = new ApiResponse<User>(true, "User logged in successfully", user);
                return Ok(response);
            }
            catch (Exception ex)
            {
                var errorResponse = new ApiResponse<string>(false, "An error occurred while logging in the user", ex.Message);
                return StatusCode(StatusCodes.Status500InternalServerError, errorResponse);
            }
        }

        [HttpPost("register")]
        public async Task<IActionResult> RegisterUser([FromBody] UserRequest userRequest)
        {
            try
            {
                var user = await _authService.RegisterUser(userRequest);
                var response = new ApiResponse<User>(true, "User created successfully", user);
                return Ok(response);
            }
            catch (Exception ex)
            {
                var errorResponse = new ApiResponse<string>(false, "An error occurred while creating the user", ex.Message);
                return StatusCode(StatusCodes.Status500InternalServerError, errorResponse);
            }
        }




    }
}

File: Server/src/controllers/authController.js
const LoginRequestDto = require("../models/requests/loginRequest");
const authService = require("../services/authService");
const ApiResponse = require("../models/responses/apiResponse"); 
const logger = require("../utils/logger");

// exports.register = async (req, res) => {
//     try {
//         const response = await authService.register(req.body);
//         res.status(201).json(ApiResponse.success("User registered successfully", response));
//     } catch (error) {
//         logger.error(`Registration Error: ${error.message}`);
//         res.status(400).json(ApiResponse.error(error.message));
//     }
// };

exports.login = async (req, res) => {
    try {
        const authRquest = new LoginRequestDto(req.body);
        const response = await authService.login(authRquest);
        res.status(200).json(ApiResponse.success("Login successful", response));
    } catch (error) {
        logger.error(`Login Error: ${error.message}`);
        res.status(400).json(ApiResponse.error(error.message));
    }
};
